relay.go

```diff
@@ -233,6 +239,17 @@ func (r *Relayer) Receive(f *Frame, fType frameType) (sent bool, failureReason s
 		return false, "relay-not-found"
 	}
 
+	finished := finishesCall(f)
+	if item.tomb {
+		// Call timed out, ignore this frame. (We've already handled stats.)
+		// TODO: metrics for late-arriving frames.
+		return true, ""
+	}
+	if finished && !item.timeout.Stop() {
+		// Timeout is firing, so no point proxying this frame
+		return true, ""
+	}
```

Comment:
Ahhhh, this makes sense. I think that this resolves an open TODO in the internal metrics code - these blocks should prevent the race we were seeing in use of the per-call metrics object.